clients/src/main/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumer.java

```diff
@@ -849,9 +953,22 @@ public OptionalLong currentLag(TopicPartition topicPartition) {
         }
     }
 
+    public void setGroupMetadata(final ConsumerGroupMetadata groupMetadata) {
+        this.groupMetadata = Optional.of(groupMetadata);
+    }
+
     @Override
     public ConsumerGroupMetadata groupMetadata() {
-        throw new KafkaException("method not implemented");
+        acquireAndEnsureOpen();
+        try {
+            maybeThrowInvalidGroupIdException();
+            backgroundEventProcessor.process();
```

Comment:
I don't think we want to process _all_ the events here, as this might process events (like async commits, et al.) that are unrelated to group metadata.

I have this same issue in another PR I'm working on, so I don't have a solution yet 😞

Reply:
OK, should then `backgroundEventProcessor.process()` only be called in `AsyncKafkaConsumer#poll()`?

Reply:
I think we should leave it only in `poll()` for the time being, yes 😞